dcpy/models/product/data_dictionary.py

```diff
@@ -0,0 +1,17 @@
+from .dataset.metadata_v2 import CustomizableBase
+from dcpy.models.base import TemplatedYamlReader
+
+
+class FieldDefinition(CustomizableBase):
+    summary: str
+    extra_description: str | None = None
+
+
+class FieldSet(CustomizableBase):
+    fields: dict[str, FieldDefinition] = {}
+
+
+class DataDictionary(CustomizableBase, TemplatedYamlReader):
+    org: dict[str, dict[str, FieldDefinition]] = {}
+    product: dict[str, dict[str, FieldDefinition]] = {}
+    dataset: dict[str, dict[str, FieldDefinition]] = {}
```

Comment:
Slowly working through this. What are your thoughts on this vs each of these being a list of `FieldSet`s, and then FieldSet could also have a `label` field. If these are analagous to tabs in an excel sheet, at first thought
```yml
datasets:
  columns:
    column_name:
      summary: column_name
      extra_description: Name of ...
  attributes:
     ...
```
is less intuitive than
```yml

datasets:
- name: column
  fields:
    column_name:
      summary: column_name
      extra_description: Name of ...
```

You can have a similar argument for the inner dictionary as well. It's definitely more of a usual pattern for our codebase, so just wanted to know what drew you to this.

Also a little curious how you imagine `org` and `product` playing in here, but can talk about that in person